packages/@azure/servicebus/data-plane/test/stress/singleMessageComplete.ts

```diff
@@ -0,0 +1,91 @@
+import { ServiceBusClient, SendableMessageInfo, OnMessage, OnError, delay } from "../../lib";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 5 * 12 * 24 * 7; // 1 week
+
+const messageMap: Set<number> = new Set<number>();
+let msgId = 1;
+
+let snapshotIntervalID: any;
+
+let isJobDone = false;
+
+async function main(): Promise<void> {
+  snapshotIntervalID = setInterval(snapshot, 5000); // Every 5 seconds
+  sendMessages();
+  receiveMessages();
```

Comment:
The above can be written as below to ensure they are run in parallel, but `main` waits for both of them
```
const sendPromise = sendMessages();
const receivePromise = receiveMessages();
return Promise.all([sendPromise, receivePromise];
```

Reply:
Discussed offline. 

Looks like we resolved related comment https://github.com/Azure/azure-sdk-for-js/pull/1887#discussion_r270589622 by mistake.

Above two versions of code are equivalent. Hence, I assumed the intent was to just use async await as it was easier to read. 
I agreed with comments about relevance of parallelism to real world scenarios, but since we had conversations about not having that for focus in P0, ignored the considerations around it,

Since we are now having discussions about it, clarifying at - https://github.com/Azure/azure-sdk-for-js/issues/1955 to avoid further PR churn. We likely have to review all async await delays, if that is a concern that we now care about.

Reply:
https://jakearchibald.com/2017/await-vs-return-vs-return-await/ is an interesting article or `return` vs `await` vs `return await`. Using just `'return' causes the rejection of promise to not be handled by the enclosing catch block if any and is usually used when we have something more to do with result returned.

For running tests as fast as possible, getting rid of some async-await delays while retaining the sequential intent of test scenario may help. Tracking these efforts in a different issue - https://github.com/Azure/azure-sdk-for-js/issues/1983

cc @mikeharder